testsuite/tests/conftest.py

```
@@ -64,9 +64,64 @@ def pytest_runtest_setup(item):
+    if report.failed:
+        report.longrepr = message
+    item.rerun_messages = getattr(item, "rerun_messages", []) + [message]
+
+    # report.sections grows with each rerun (attempt1 + attempt2 + ...).
+    # Slice from where the previous attempt ended to isolate this attempt's output.
+    captured_output = ""
+    for _, section_content in report.sections:
+        if section_content:
+            captured_output += section_content
+    prev_output = getattr(item, "rerun_prev_output", "")
+    item.rerun_outputs = getattr(item, "rerun_outputs", []) + [captured_output[len(prev_output) :]]
```

Ruff (0.15.10)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)